src/main/java/Duke.java

```diff
@@ -1,10 +1,190 @@
+import java.util.Scanner;
+//import java.util.Arrays;
+
 public class Duke {
+
+    private static final Scanner input = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String userInput = "";
+        Task[] currentTasks = new Task[100];
+        int numOfTasks = 0;
+
+        showWelcomeMessage();
+
+        while(!userInput.equals("bye")) {
+            userInput = getUserInput();
+            String[] splitUserInput = splitCommands(userInput);
+
+            switch (splitUserInput[0]) {
+
+                case "bye":
+                    exitProgram();
+                    break;
+
+                case "list":
+                    commandList(currentTasks, numOfTasks);
+                    break;
+
+                case "done":
+                    markAsDone(currentTasks, numOfTasks, splitUserInput[1]);
+                    break;
+
+                case "deadline":
+                    numOfTasks = commandDeadline(currentTasks, numOfTasks, splitUserInput[1]);
+                    break;
+
+                case "todo":
+                    numOfTasks = commandToDo(currentTasks, numOfTasks, splitUserInput);
+                    break;
+
+                case "event":
+                    numOfTasks = commandEvent(currentTasks, numOfTasks, splitUserInput[1]);
+                    break;
+
+                case "help":
+                    commandHelp();
+                    break;
+
+                default:
+                    invalidCommand();
+            }
+        }
     }
+
+
+
+    /*
+     * ===========================================
+     *           MAIN COMMANDS
+     * ===========================================
+     */
+    private static void commandList(Task[] currentTasks, int numOfTasks) {
+        printSeparator();
+        for(int i = 0; i < numOfTasks; i++){
+            System.out.println((i+1) + "." +addBrackets(currentTasks[i].getType()) +
+                    addBrackets(currentTasks[i].getStatusIcon()) + " " + currentTasks[i].getDescription());
+        }
+        printSeparator();
+    }
+
+    private static int commandDeadline(Task[] currentTasks, int numOfTasks, String s) {
+        String[] splitInfoAndDeadline = s.trim().split("/by",2 );
```

Comment:
Is it possible to remove magic number, try declaring  "/by" as a constant DELIMIT_DEADLINE.

Reply:
Alright, will do that in my next commit!